### ISSUE TITLE
Incorporate Knowledge Distillation

### ISSUE THREAD

**Issue by bigaidream** [bigaidream-projects/qan#11]
To test if KD will help further accelerate the tuning process. 

1. Train a CNN in a normal way and save the weights. 
2. Use knowledge distillation to modify the training data's labels. 
3. Apply QAN on the modified data.

**Comment by bigaidream** [comment]
The reference code can be found here: https://github.com/adri-romsor/FitNets

**Comment by bigaidream** [comment]
Torch code for knowledge distillation: https://github.com/natoromano/specialistnets

**Comment by bigaidream** [comment]
It seems that `meta-momentum` does most of the job of `knowledge distillation`. In other words, KD seems not very helpful on small datasets. But maybe KD can still help in ImageNet?
